core/src/main/scala/org/apache/spark/util/SizeEstimator.scala

```diff
@@ -73,7 +74,8 @@ object SizeEstimator extends Logging {
   private val ALIGN_SIZE = 8
 
   // A cache of ClassInfo objects for each class
-  private val classInfos = new ConcurrentHashMap[Class[_], ClassInfo]
+  // We use weakKeys to allow GC of dynamically created classes
+  private val classInfos = new MapMaker().weakKeys().makeMap[Class[_], ClassInfo]()
```

Comment:
A WeakHashMap would need to be synchronized and we would therefore lose concurrency.
I don't know if the SizeEstimator is used concurrently and hence felt it was safer to keep the concurrency of the original code despite the added complexity of a Guava Map.

Is it safe to drop the concurrency requirement?